### DISCUSSION TITLE
Adding dynamic parameters : 02. PyTorch Neural Network Classification - 6.2 Building a model with non-linearity

### DISCUSSION THREAD

**Discussion by vinodarya** [mrdbourke/pytorch-deep-learning#1377] [Ideas]
Hi Daniel

Thanks a lot for your wonderful tutorials, https://www.learnpytorch.io/02_pytorch_classification/#62-building-a-model-with-non-linearity

I have enhanced the code to add dynamic creation of the model based on the following parameters inputted by the User in the Form:

a. Number of epochs
b. Number of hidden layer
c. Linearity enabled or not

The NN model is created based on the parameters and when non-linearity is enabled the Model achieves high accuracy.

Path to my notebook, inspired from your codebase 
https://github.com/vinodarya/machinelearning/blob/main/PyTorch_NN_playground%20v1.ipynb

Thanks
Vinod Arya

Sample run
<img width="634" height="500" alt="image" src="https://github.com/user-attachments/assets/5e59a64d-05ff-4278-b8d5-2d930ca10033" />
